src/jit/emitxarch.cpp

```diff
@@ -1625,13 +1628,20 @@ inline emitter::code_t emitter::insEncodeReg3456(instruction ins, regNumber reg,
     assert(IsAVXInstruction(ins));
     assert(hasVexPrefix(code));
 
-    // Get 4-bit register encoding
-    // RegEncoding() gives lower 3 bits
-    // IsExtendedReg() gives MSB.
-    code_t regBits = RegEncoding(reg);
-    if (IsExtendedReg(reg))
+    code_t regBits = 0;
+
+    // VEX.vvvv bits are reserved for some instructions
+    // and must be set to 1111 otherwise CPU will #UD
+    if (ins != INS_vpermpd && ins != INS_vpermq)
```

Comment:
I'm not positive this is the right fix. `vpermpd` and `vpermq` are, by far, not the only instructions which must set VEX.vvvv bits to 1111b and many of them (like `comisd`) have been around and in use for quite some time.